pkg/proxy/iptables/proxier.go

```diff
@@ -552,6 +572,61 @@ func servicePortEndpointChainName(s proxy.ServicePortName, protocol string, endp
 	return utiliptables.Chain("KUBE-SEP-" + encoded[:16])
 }
 
+// getRemovedEndpointIPs returns the endpoint IPs that are missing in the new endpoints
+func getRemovedEndpoints(curEndpoints, newEndpoints []string) []string {
+	var result []string
+	epSet := sets.NewString(newEndpoints...)
+	for _, ep := range curEndpoints {
+		if epSet.Has(ep) {
```

Comment:
don't you want the opposite of this?  If the new set DOESN'T have the ep it is removed.  Or am I misunderstanding this?